### DISCUSSION TITLE
Wide table or multiple tables: How manage related and repeated data?

### DISCUSSION THREAD

**Discussion by info2000** [ClickHouse/ClickHouse#42207] [Q&A] [answered]
On an ecommerce pageviews case, where each product pageview, has the user and product properties (+100 values, most of the varchar 200) + the pageview properies (~10 values)

What is better in this case:
a)had 1 table with 110 columns, and around 80% of the stored data is repeated 

b)had 3 tables: pageviews with 12-15 columns  (add the most used columns from user and products, like country ie) and create a materialized view to show the pageviews data with the joined data from user and products

the most important query (3 times per week), will need the 80 of the total 110 values
The most used queries will aggregations of data grouped by some (3-12) user and products properties

I readed a lot of blog posts and also asked to some "clickhouse users" but didn't find an experienced answer.

Thanks

**Comment by den-crane** [comment, marked as answer]
>a)had 1 table with 110 columns, and around 80% of the stored data is repeated

This option is better.

**Reply by info2000** [reply]
Many thanks!
